x-pack/plugins/security_solution_serverless/public/get_started/card_item.tsx

```diff
@@ -5,144 +5,115 @@
   CardId,
   ExpandedCardSteps,
-  OnCardClicked,
-  OnStepButtonClicked,
+  ToggleTaskCompleteStatus,
   OnStepClicked,
   SectionId,
   StepId,
 } from './types';
-import * as i18n from './translations';
-import { CardStep } from './card_step';
 import { getCard } from './helpers';
 import type { ProductLine } from '../../common/product';
+import { CardStep } from './card_step';
+
+const SHADOW_ANIMATION_DURATION = 350;
 
 const CardItemComponent: React.FC<{
   activeProducts: Set<ProductLine>;
   activeStepIds: StepId[] | undefined;
   cardId: CardId;
   euiTheme: EuiThemeComputed;
   expandedCardSteps: ExpandedCardSteps;
-  finishedSteps: Record<CardId, Set<StepId>>;
-  onCardClicked: OnCardClicked;
-  onStepButtonClicked: OnStepButtonClicked;
+  finishedSteps: Set<StepId>;
+  toggleTaskCompleteStatus: ToggleTaskCompleteStatus;
   onStepClicked: OnStepClicked;
   sectionId: SectionId;
-  shadow?: string;
-  stepsLeft?: number;
-  timeInMins?: number;
 }> = ({
   activeProducts,
   activeStepIds,
   cardId,
   euiTheme,
   expandedCardSteps,
   finishedSteps,
-  onCardClicked,
-  onStepButtonClicked,
+  toggleTaskCompleteStatus,
   onStepClicked,
   sectionId,
-  shadow,
-  stepsLeft,
-  timeInMins,
 }) => {
+  const isExpandedCard = expandedCardSteps[cardId].isExpanded;
+
   const cardItem = useMemo(() => getCard({ cardId, sectionId }), [cardId, sectionId]);
-  const expandCard = expandedCardSteps[cardId]?.isExpanded ?? false;
   const expandedSteps = useMemo(
     () => new Set(expandedCardSteps[cardId]?.expandedSteps ?? []),
     [cardId, expandedCardSteps]
   );
-  const toggleCard = useCallback(
-    (e) => {
-      e.preventDefault();
-      const isExpanded = !expandCard;
-      onCardClicked({ cardId, isExpanded });
-    },
-    [cardId, expandCard, onCardClicked]
-  );
-  const hasActiveSteps = activeStepIds != null && activeStepIds.length > 0;
-  return cardItem && hasActiveSteps ? (
+
+  const shadow = useEuiShadow('l');
+  const cardClassNames = classnames({
+    'card-expanded': isExpandedCard,
+  });
+
+  const iconHoveredBackgroundColor = useEuiBackgroundColor('success');
+
+  return cardItem && activeStepIds ? (
     <EuiPanel
+      className={cardClassNames}
       hasBorder
       paddingSize="none"
       css={css`
-        ${shadow ?? ''};
-        padding: ${euiTheme.size.base} ${euiTheme.size.l} ${euiTheme.size.l};
+        padding: ${euiTheme.size.base};
         margin-bottom: ${euiTheme.size.xs};
-        border-radius: ${euiTheme.size.xs};
+        border-radius: ${euiTheme.size.s};
+        border: 1px solid ${euiTheme.colors.lightShade};
+        box-sizing: content-box;
+
+        &:hover,
+        &.card-expanded {
+          ${shadow};
+          transition: box-shadow ${SHADOW_ANIMATION_DURATION}ms ease-out;
+
+          & .step-icon {
+            background-color: ${iconHoveredBackgroundColor};
+          }
+        }
+
+        &.card-expanded {
+          border: 2px solid #6092c0;
+        }
```

Comment:
Can we separate all this CSS from the component implementation, in a `usePanelStyles` hook or something? Also, why do we ask for the `euiTheme` to be passed by props? Can call `useEuiTheme` where we need it?